packages/core/src/model.ts

```diff
@@ -116,6 +114,19 @@ export class ChartModel {
 	 */
 	setData(newData) {
 		const sanitizedData = this.sanitize(Tools.clone(newData));
+
+		// Sort data in user defined order
+		const legendOrder = Tools.getProperty(
+			this.getOptions(),
+			"legend",
+			"order"
+		);
+		if (legendOrder.length) {
+			sanitizedData.sort((dataA, dataB) => 
+				legendOrder.indexOf(dataA.group) - legendOrder.indexOf(dataB.group)
```

Comment:
Will this work/should it account for duplicate labels? Like let's say `legendOrder` is `["Restocking", "Misc","Sold", "Qty", "More", "Restocking"]`.

Reply:
I don't think we need to worry about duplicates in the legend, but interested to see what order you'd get from that array

Reply:
I just tested. It doesn't make any difference with duplicates.